kbase-extension/static/kbase/js/widgets/narrative_core/kbaseAppCard.js

```diff
@@ -114,21 +113,7 @@ define([
                 },
             });
 
-        const $logo = $('<div>');
-        if (app.icon && app.icon.url) {
-            const url = Config.url('narrative_method_store_image') + app.icon.url;
-            $logo.append(
-                DisplayUtil.getAppIcon({
-                    url: url,
-                    cursor: 'pointer',
-                    setColor: true,
-                    size: '50px',
-                })
-            );
-        } else {
-            $logo.append(DisplayUtil.getAppIcon({ cursor: 'pointer', setColor: true }));
-        }
-
+        const $logo = $('<div>').append(Icon.makeAppIcon(entry.app));
```

Comment:
From some work I did awhile back on centralising all the Icon functions to the same file.